### DISCUSSION TITLE
Question on Password-Based auth

### DISCUSSION THREAD

**Discussion by wenjunche** [emqx/emqx#12965] [General]
Hi

We are testing EMQX 5.3.1 (docker based free version),  and have JWT and Password-Based (built-in DB) configured.   We noticed, if `mqtt.connectAsync` is called with invalid username or invalid password,  it returns `Connection refused: Not authorized`, as expected.   However,  if it called with any username without password,  it is able to connect, and Dashboard->Client shows the connection with a 'random' string as username.  How can I make password required when connecting through Password-Based auth ?

Thanks in advance

**Comment by thalesmg** [comment]
Hi.

Could you please clarify where does this `mqtt.connectAsync` configuration comes from?  It's not a broker configuration.  Sounds like something in your MQTT client?

I was not able to reproduce the issue you mentioned about anonymous connections.  When an authenticator is configured and enabled, MQTT clients without username should be denied connection.

Could you try a newer version of EMQX (v5.6.1 is the latest at the time of writing) and see if you can reproduce the issue there?
